integrationTests/singleShard/block/executingMiniblocksSc_test.go

```diff
@@ -0,0 +1,342 @@
+var stepDelay = time.Second
+
+func TestShouldProcessBlocksWithScTxsJoinAndRewardTheOwner(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	log := logger.DefaultLogger()
+	log.SetLevel(logger.LogDebug)
+
+	scCode, err := ioutil.ReadFile(agarioFile)
+	assert.Nil(t, err)
+
+	maxShards := uint32(1)
+	numOfNodes := 4
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+	advertiserAddr := integrationTests.GetConnectableAddress(advertiser)
+
+	nodes := make([]*integrationTests.TestProcessorNode, numOfNodes)
+	for i := 0; i < numOfNodes; i++ {
+		nodes[i] = integrationTests.NewTestProcessorNode(maxShards, 0, 0, advertiserAddr)
+	}
+
+	idxProposer := 0
+	hardCodedSk, _ := hex.DecodeString("5561d28b0d89fa425bbbf9e49a018b5d1e4a462c03d2efce60faf9ddece2af06")
+	hardCodedScResultingAddress, _ := hex.DecodeString("000000000000000000005fed9c659422cd8429ce92f8973bba2a9fb51e0eb3a1")
+	nodes[idxProposer].LoadTxSignSkBytes(hardCodedSk)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Messenger.Close()
+		}
+	}()
+
+	for _, n := range nodes {
+		_ = n.Messenger.Bootstrap()
+	}
+
+	fmt.Println("Delaying for nodes p2p bootstrap...")
+	time.Sleep(stepDelay)
+
+	round := uint32(0)
+	round = incrementAndPrintRound(round)
+
+	initialVal := big.NewInt(10000000)
+	topUpValue := big.NewInt(500)
+	withdrawValue := big.NewInt(10)
+	mintAllNodes(nodes, initialVal)
+
+	deployScTx(nodes, idxProposer, string(scCode))
+	proposeBlock(nodes, idxProposer, round)
+	syncBlock(t, nodes, idxProposer, round)
+	round = incrementAndPrintRound(round)
+
+	nodeDoesJoinGame(nodes, idxProposer, topUpValue, hardCodedScResultingAddress)
+	proposeBlock(nodes, idxProposer, round)
+	syncBlock(t, nodes, idxProposer, round)
+	round = incrementAndPrintRound(round)
+
+	checkJoinGameIsDoneCorrectly(
+		t,
+		nodes,
+		idxProposer,
+		idxProposer,
+		initialVal,
+		topUpValue,
+		hardCodedScResultingAddress,
+	)
+
+	nodeCallsRewardAndSend(nodes, idxProposer, idxProposer, withdrawValue, hardCodedScResultingAddress)
+	proposeBlock(nodes, idxProposer, round)
+	syncBlock(t, nodes, idxProposer, round)
+	round = incrementAndPrintRound(round)
+
+	proposeBlock(nodes, idxProposer, round)
+	syncBlock(t, nodes, idxProposer, round)
+	round = incrementAndPrintRound(round)
+
+	checkRewardIsDoneCorrectly(
+		t,
+		nodes,
+		idxProposer,
+		idxProposer,
+		initialVal,
+		topUpValue,
+		withdrawValue,
+		hardCodedScResultingAddress,
+	)
+
+	checkRootHashes(t, nodes, []int{0})
+
+	time.Sleep(1 * time.Second)
+}
+
+func incrementAndPrintRound(round uint32) uint32 {
+	round++
+	fmt.Printf("#################################### ROUND %d BEGINS ####################################\n\n", round)
+
+	return round
+}
+
+func mintAllNodes(nodes []*integrationTests.TestProcessorNode, value *big.Int) {
+	for _, n := range nodes {
+		if n.ShardCoordinator.SelfId() == sharding.MetachainShardId {
```

Comment:
delete this if, it does not make sense

Reply:
actually it will break the code otherwise because metachain nodes will try to mint addresses